pkg/internal/coordinatortest/marshaling.go

```diff
@@ -0,0 +1,255 @@
+package coordinatortest
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/coordinator"
+	"github.com/keep-network/keep-core/pkg/internal/hexutils"
+	"github.com/keep-network/keep-core/pkg/tbtc"
+)
+
+// UnmarshalJSON implements a custom JSON unmarshaling logic to produce a
```

Comment:
Just a thought that maybe it would make more sense to get rid of the JSON-unmarshaling code and represent test scenarios in `.go` files using the structures defined in `coordinatortest.go`? We can define inline lists and inline maps in Go. I don't hate the current approach, just have mixed feelings if it's worth to have all the Go code from `pkg/internal/coordinatortest` compared to just having structs and inline definitions of test scenarios in Go. 

Example:
```
var AutoSelectWalletAndFilterOutSweptDeposits = FindDepositsToSweepTestScenario{
	Title:               "Auto-select wallet and filter out swept deposits",
	MaxNumberOfDeposits: 3,
	WalletPublicKeyHash: "",
	Wallets: []Wallet {
		WalletPublicKeyHash: hexToPubkey("0x928d992e5f5b71de51a1b40fcc4056b99a88a647"),
		RegistrationBlockNumber: 5,
	  },
	  {
		WalletPublicKeyHash: hexToPubkey("0x03b74d6893ad46dfdd01b9e0e3b3385f4fce2d1e"),
		RegistrationBlockNumber: 10,
	  },
	  {
		WalletPublicKeyHash: hexToPubkey("0x7670343fc00ccc2d0cd65360e6ad400697ea0fed"),
		RegistrationBlockNumber: 20,
	  },
	}
	Deposits: []Deposit {
		Deposit {
			RevealBlockNumber: 6,
			WalletPublicKeyHash: hexToPubkey("0x928d992e5f5b71de51a1b40fcc4056b99a88a647"),
			FundingTxHash: hashFromString("dbc6166e149995ef4ed6428d23703756971256ab19954e099fdf2a96fa332251"),
			FundingOutputIndex: 1,
			FundingTxConfirmations: 20,
			SweptAt: 102
		}
	},
   // (...)
}
```

`hexToPubkey` and `hashFromString` would only be defined for tests and they would panic in case some problems with the input.

What do you think? Non-blocking.

Reply:
It makes sense, we use a similar approach in `internal/testdata/bitcoin/transaction.go`.
Here I wanted to be consistent with `pkg/internal/tbtctest` though. We should align them all.

Reply:
I also think the main differentiator is the size of the test data. For complex structures, I think it's better to use separate JSON files. For simple ones, we can stick with the inline approach. Not sure if we should aim for aligning them at all costs.